From c459043572f18e6df9128a7418187d2a69cdc393 Mon Sep 17 00:00:00 2001
From: 7Ji <pugokushin@gmail.com>
Date: Wed, 26 Mar 2025 15:41:53 +0800
Subject: [PATCH] wayland: never use implicit sync

Current chromium logic is to use it for kernels that "supposedly" supports
implicit sync, that is, any kernel >= 6.0. However, our BSP kernels never
follow the mainline feature set, and implicit sync is yet another
feature they do not support.

Disable it always by making function
ui::WaylandConnection::UseImplicitSyncInterop always return false  so
we touch as few codes as possible, and hopefully clang would inline and
kill the dead codes.
---
 ui/ozone/platform/wayland/host/wayland_connection.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/ui/ozone/platform/wayland/host/wayland_connection.h b/ui/ozone/platform/wayland/host/wayland_connection.h
index 6466f90..58ca04c 100644
--- a/ui/ozone/platform/wayland/host/wayland_connection.h
+++ b/ui/ozone/platform/wayland/host/wayland_connection.h
@@ -359,8 +359,7 @@ class WaylandConnection {
   void DumpState(std::ostream& out) const;
 
   bool UseImplicitSyncInterop() const {
-    return !SupportsExplicitSync() &&
-           WaylandBufferManagerHost::SupportsImplicitSyncInterop();
+    return false;
   }
 
   // Returns a sync callback, which is invoked when the server has processed all
-- 
2.49.0

